### PR TITLE
Allowing FPS to take numpy array of ints as initialize parameter

### PR DESCRIPTION
**Pull request by cajchristian**
Hello!

This pull request is my attempt to address issue #145. With these changes, FPS should be able to accept a numpy array of ints as an input. I wrote a few tests as well to ensure the updated code is functioning properly. I'd love to hear any feedback you may have!

Contributor (creator of PR) checklist
-------------------------------------
 - [x] Tests updated (for new features and bugfixes)?
 - [x] Documentation updated (for new features)?
 - [x] Issue referenced (for PRs that solve an issue)?

For Reviewer
------------
 - [ ] CHANGELOG updated if important change?

----
📚 Documentation preview 📚: https://scikit-matter--225.org.readthedocs.build/en/225/

### CODE REVIEW

File: src/skmatter/_selection.py [hunk 1]
```diff
@@ -1038,7 +1038,14 @@ def _init_greedy_search(self, X, y, n_to_select):
...
         self.hausdorff_at_select_ = np.full(X.shape[self._axis], np.inf)
 
-        if self.initialize == "random":
+        if isinstance(self.initialize, np.ndarray):
+            if all(isinstance(i, numbers.Integral) for i in self.initialize):
+                for i, val in enumerate(self.initialize):
+                    self.selected_idx_[i] = val
+                    self._update_post_selection(X, y, self.selected_idx_[i])
+            else:
+                raise ValueError("Initialize parameter must contain only int")
```
Comment (PicoCentauri) [inline src/skmatter/_selection.py:1047]: Would you mind adding a test for this error raise?

You can take a look how to test that in the [pytest docs](https://docs.pytest.org/en/stable/reference/reference.html#pytest.raises) or find another example in our test suite

Comment (cajchristian) [reply src/skmatter/_selection.py]: Just added a test for the error raise based on the other tests in test_feature_simple_fps.py. Let me know if this looks good to you!

Comment (cajchristian) [reply src/skmatter/_selection.py]: Should be fixed now. Added the unit tests I wrote in test_feature_simple_fps to test_sample_simple_fps. Let me know if this looks correct!

File: src/skmatter/_selection.py [hunk 2]
```diff
@@ -1038,7 +1038,14 @@ def _init_greedy_search(self, X, y, n_to_select):
         self.hausdorff_ = np.full(X.shape[self._axis], np.inf)
         self.hausdorff_at_select_ = np.full(X.shape[self._axis], np.inf)
 
-        if self.initialize == "random":
+        if isinstance(self.initialize, np.ndarray):
```
Comment (PicoCentauri) [inline src/skmatter/_selection.py:1041]: Shouldn't these operattions also be performed for a List and not only for a numpy array?

Comment (cajchristian) [reply src/skmatter/_selection.py]: Yes. Let me know what you think about the new changes I made.

**Review by PicoCentauri — COMMENTED** [review]
Yes looks good. Can you add to the [CHANGELOG](https://github.com/scikit-learn-contrib/scikit-matter/blob/main/CHANGELOG)?

**Comment by cajchristian** [comment]
Updated the changelog!

**Review by PicoCentauri — APPROVED** [review]
Very nice. Thanks for this addition!

**Pull request merged**